vello_hybrid/src/render.rs

```diff
@@ -161,6 +201,133 @@ impl CsRenderCtx {
+            }
+            if tile_y == clip_bbox[3] {
+                break;
+            }
+            let x0 = strip.x() as usize;
+            let x_clamped = (x0 / WIDE_TILE_WIDTH).min(clip_bbox[2]);
+            if tile_x < x_clamped {
+                if core::mem::take(&mut pop_pending) {
+                    self.tiles[tile_y * width_tiles + tile_x].pop_clip();
+                    tile_x += 1;
+                }
+                // The winding check is probably not needed; if there was a fill,
+                // the logic below should have advanced tile_x.
+                if strip.winding == 0 {
+                    for x in tile_x..x_clamped {
+                        self.tiles[tile_y * width_tiles + x].pop_zero_clip();
+                    }
+                }
+                tile_x = x_clamped;
+            }
+            let next_strip = &strips[i + 1];
+            let strip_width = (next_strip.col - strip.col) as usize;
+            let x1 = x0 + strip_width;
+            let xtile0 = (x0 / WIDE_TILE_WIDTH).max(clip_bbox[0]);
+            let xtile1 = x1.div_ceil(WIDE_TILE_WIDTH).min(clip_bbox[2]);
+            let mut x = x0;
+            let mut alpha_ix = strip.col as usize;
+            if clip_bbox[0] * WIDE_TILE_WIDTH > x {
+                alpha_ix += clip_bbox[0] * WIDE_TILE_WIDTH - x;
+                x = clip_bbox[0] * WIDE_TILE_WIDTH;
+            }
+            for xtile in xtile0..xtile1 {
+                if xtile > tile_x && core::mem::take(&mut pop_pending) {
+                    self.tiles[tile_y * width_tiles + tile_x].pop_clip();
+                }
+                let x_tile_rel = (x % WIDE_TILE_WIDTH) as u32;
+                let width = x1.min((xtile + 1) * WIDE_TILE_WIDTH) - x;
+                let cmd = CmdClipStrip {
+                    x: x_tile_rel,
+                    width: width as u32,
+                    alpha_ix,
+                };
+                x += width;
+                alpha_ix += width;
+                self.tiles[tile_y * width_tiles + xtile].clip_strip(cmd);
+                tile_x = xtile;
+                pop_pending = true;
+            }
+            if next_strip.winding != 0 && y == next_strip.strip_y() as usize {
+                let x2 = next_strip.x() as usize;
+                let tile_x2 = x2.min((tile_x + 1) * WIDE_TILE_WIDTH);
+                let width = tile_x2 - x1;
+                if width > 0 {
+                    let x_tile_rel = (x1 % WIDE_TILE_WIDTH) as u32;
+                    self.tiles[tile_y * width_tiles + tile_x].clip_fill(x_tile_rel, width as u32);
+                }
+                if x2 > (tile_x + 1) * WIDE_TILE_WIDTH {
+                    self.tiles[tile_y * width_tiles + tile_x].pop_clip();
+                    let width2 = x2 % WIDE_TILE_WIDTH;
+                    tile_x = x2 / WIDE_TILE_WIDTH;
+                    if width2 > 0 {
+                        self.tiles[tile_y * width_tiles + tile_x].clip_fill(0, width2 as u32);
```

Comment:
AFAICT, tile_x might exceed the width of the image here, for example when clipping to a rectangle that is wider than the image.